drivers/flash/soc_flash_nrf.c

```diff
@@ -137,6 +144,34 @@ static int flash_nrf_read(const struct device *dev, off_t addr,
 		return 0;
 	}
 
+	/* If the currently executing image is non-secure and the address
+	 * is not in the non-secure image we will not be able to read it
+	 * directly and must instead invoke a secure service to read out
+	 * the flash.
+	 */
+#ifdef CONFIG_ARM_NONSECURE_FIRMWARE
```

Comment:
Should use `CONFIG_BUILD_WITH_TFM` here, similar to above where the header is included.

Reply:
@joerchan, won't this change result in also calling the `tfm_platform_mem_read()` when performing a read from a secure firmware ?